packages/edit-site/src/components/theme-exporter/index.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * External dependencies
+ */
+import { saveAs } from 'file-saver';
+
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+import { useCallback } from '@wordpress/element';
+import { MenuItem } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import getBaseThemeZip from './get-base-theme-zip';
+
+export default function ThemeExporter( { ids, templatePartIds } ) {
```

Comment:
This PR is doing too much IMO, can't we keep the exported for a separate PR.

Reply:
I thought it was ok because it's a stand-alone prototype and doesn't change public APIs.

Reply:
The PR is too big and harder to test, review and land. It's not about APIs it's about landing small pieces iteratively.

Reply:
I'll break this one up once it gets pseudo-approved, but I worry we are falling for a dogmatic requirement of keeping PRs small.

There is a higher up-front cost for review sure, but you save on the overhead of stacking PRs, and complex rebases every time a significant change happens on the master branch, or you finally get a review, and one of the base branches is merged.

In general, I agree that smaller PRs tend to be better. Still, there is a place for PRs like this when introducing an entirely new module that doesn't modify existing code, especially when the set of features introduced are almost all sequential. If they were things that could be merged parallelly, I would see value in a parallel approach. Otherwise, you're increasing the total person-hours to get the same code in with the same scrutiny, or less, if you consider different people could review the various sequential PRs without the best context of the previous and future changes.

Reply:
I have to disagree, loading multiple entities, and exporting entities seems like two complete different features to me and are better addressed in separate PRs. (no dogme here)

Reply:
I wasn't thinking about specific examples, but you can't test exporting a theme without loading the templates.